### ISSUE TITLE
Missing package.json?

### ISSUE THREAD

**Issue by moritonal** [kakaroto/Beyond20#296]
Hi @kakaroto, I was looking to provide a PR for the permissions issue #179  but I ran into the issue that there isn't a `package.json` in the main to describe the development process.

Rough gist is that the project might be improved by adding a "package.json" for main repo and adding commands for `npm run build-firefox` or `npm run build-chrome`, along with allowing `npm install` to simply install the dev-dependencies.

It might make more sense to compile all the files into a "./build/firefox" and "./dist/chrome" to avoid having to rename files and makes things more obvious.

I'll look into this happily, just wanted to raise it so I can get your feedback and thoughts.

**Comment by kakaroto** [comment]
Hi Tom,
Thanks for the help!
I don't really see the point of having a package.json, but it might just be because I'm used to doing things differently, and this isn't a node application, so I'm not sure how why it would be necessary. The build process has changed recently and I haven't gone through the release process with the new build system yet.
The README has instructions though on how to build it. You need to have `gulp` installed and just run `gulp` to build the dist files.

The build is done with 'make build' which runs the webext program (which I then modify manually the generated zip to remove the extraneous stuff from the docs folder and the src/ dir), and the only difference between firefox and chrome is the `manifest_ff.json` is renamed to `manifest.json`.

I suppose one run command that the package.json could have is for launching the standalone docs server, I have a run.sh file in the docs folder with this in it :
`~/bin/bundle exec jekyll serve --incremental --host 0.0.0.0`
I don't even remember what `bundle` is or where it came from, but it works and I don't touch it once it works 🤷 
If you can improve this process, that'd be great!

**Comment by moritonal** [comment]
So that makes sense, but can I suggest a more developed build pipeline? I see now there's a Make file, but honestly I didn't even notice that when I first started looked at this project, and have no idea how I'd run it on a windows box.

You already have gulp (which means we can assume node is installed) which means we could use that build a much clearer process?

If you look at the PR (#300) I submitted you'll see I've used a `package.json` to wrap up all the dependencies, build-process and other meta things, which make the process of building and packaging a version for Chrome and Firefox as simple as:

`npm install`
`npm run build`
`npm run package`

Along with things like `npm run start:chrome` for development. It's in no ways perfect, but I've used it many times before on projects and thought you might be interested in seeing it.
